corebc-core/src/types/ens.rs

```diff
@@ -102,7 +102,7 @@ impl FromStr for NameOrAddress {
     type Err = <Address as FromStr>::Err;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        if s.starts_with("0x") {
+        if (s.len() == 44 || s.len() == 46)  && (s.starts_with("cb") || s.starts_with("ab") || s.starts_with("ce")) {
```

Comment:
Here what if the len will be 46? Then the start will be `0x` not one of the options. I think we will have to add `0x` as one of the options.

Reply:
can you check pls now?